### ISSUE TITLE
Control the SNAT using CRD

### ISSUE THREAD

**Issue by liorfranko** [F5Networks/k8s-bigip-ctlr#1397]
#### Title
We would like to be able to disable the SNAT using the configmap

#### Description
We're using the bigip-ctlr to manage K8S SVC objects using configmaps.
All the VIPs on the F5 are created with "Source Address Translation - Automap" without the ability to change it.
We need to control it, on some VIPs we need to change it to None.
The new way of deployment using CRD doesn't provide all the existing features, for example:

- When using the CRD, the pool is not attached to the VIP but uses a Policy, this change reduces the visibility level we have now, and we won't be able to see the status of the VIPs.
- After an object is created using a CRD, the bigip-ctlr stop enforcing the VIP's configuration and allow changes to be made on the F5 UI.
- The VIP name on the F5 contains only the allocated IP address and doesn't provide a way to modify it in the CRD.
- Migrating from the configmap to the CRD will take time, the use of CRD is not just a cosmetic change but a major change that needs to be rollout carefully.

#### Actual Problem
The SNAT parameter is hardcoded in the configmap

#### Solution Proposed
Add a flag called SNAT to the configmap, that can take "Automap" or "None" as values.

**Comment by cisbotctlr** [comment]
cisbot will assign the issue to one of the devs. 
 @devs, use /jira for internal tracking.

**Comment by mdditt2000** [comment]
@liorfranko thank you opening this. Will discuss with CIS team at next week. The ask is very clear.

**Comment by mdditt2000** [comment]
Option to add SNAT, No SNAT or use a SNAT policy has been added to CRDs and will be released in the CIS 2.2 release. 

Example:

```
Creating an additional parameter in Virtual resource definition as follows:

apiVersion: cis.f5.com/v1
kind: VirtualServer
metadata:
{{ labels:}}
{{ f5cr: "true"}}
{{ name: coffee-virtual-server}}
{{ namespace: default}}
spec:
snat: /Common/snat 
{{ tlsProfileName: reencrypt-tls}}
{{ host: coffee.example.com}}
{{ pools:}}
{{ - path: /lattee}}
{{ service: svc}}
{{ servicePort: 80}}
{{ - path: /mocha1}}
{{ service: svc}}
{{ servicePort: 80}}
{{ virtualServerAddress: 172.16.3.5}}
```

Pull request   https://github.com/F5Networks/k8s-bigip-ctlr/pull/1487

Image is available to test if requested.

**Comment by mdditt2000** [comment]
Good morning @liorfranko @pcloup  I have a CRD build that supports no snat option. Here is the schema

https://github.com/F5Networks/k8s-bigip-ctlr/blob/master/docs/_static/config_examples/crd/Install/customresourcedefinitions.yml

CRDs also support Node label for NodePort and a service with multiple ports. Can i provide you a build for testing?
